src/sentry/eventstream/kafka/protocol.py

```diff
@@ -172,9 +190,27 @@ def get_task_kwargs_for_message_from_headers(
                 "is_regression": is_regression,
                 "is_new_group_environment": is_new_group_environment,
             }
```

Comment:
Actually wondering if `group_states` should be included here instead? nbd either way

Reply:
It might make sense to put it in there since it is apart of 'task state'. Are we planning to keep the flattened out params `is_new`, `is_regression`, and `is_new_group_environment` that are used for error issues as well as the multiple group states all in the task signature? 

If we keep `group_states` separate from `task_state` it'll be easier to do cleanup.